packages/react-app/src/contexts/BridgeContext.jsx

```diff
@@ -73,9 +73,11 @@ export const BridgeProvider = ({ children }) => {
       maxPerTx: defaultMaxPerTx(token.decimals),
       dailyLimit: defaultDailyLimit(token.decimals),
     });
-    fetchTokenLimits(token).then(limits => {
-      setTokenLimits(limits);
-    });
+    if (providerNetwork && token.chainId === providerNetwork.chainId) {
+      fetchTokenLimits(token, ethersProvider).then(limits => {
+        setTokenLimits(limits);
+      });
+    }
```

Comment:
so i just replaced the networkMismatch by rechecking the same logic against the token.chainId since we already have those details here. instead of relying on the `setNetworkMismatch`, it will just recheck again and again until the provider network is available. If no wallet is connected, it does not run this piece.